TicketBooking/AppSettings/AppSettingsInitializer.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace TicketBooking.AppSettings;
+
+public class AppSettingsInitializer
+{
+    private static AppSettingsModel? _appSettings;
+
+    public static AppSettingsModel AppSettingsInstance
+    {
+        get
+        {
+            if (_appSettings is null)
+            {
+                string json = File.ReadAllText("..\\..\\..\\AppSettings\\appsettings.json");
+                _appSettings = JsonSerializer.Deserialize<AppSettingsModel>(json);
+            }
+
+            return _appSettings;
+        }
+    }
```

Comment:
I'm trying to apply singleton pattern for appsettings instance, but the implementation you suggested doesn't
Talking about the entire implementation, not the point of changing it to static method

Reply:
@raghad-ns Singletons and static classes share a lot of properties. I personally prefer to have constant values in static classes over singletons, however both can achieve the same goals and most of the time can be used interchangeably. If you want to achieve Singleton behavior in this class, I recommend setting the data of _appSettings in a private constructor, so _appSettings can never be null and your singleton won't be instantiated. This has two advantages over doing it on the `get` accessor: 

1. A user of your class doesn't need to concern themselves with the `get` accessor having side effects (in this case, loading the settings from a file in case its null) and multiple responsibilities. 
2. You will avoid the if check every time

Reply:
Btw, upper code is an example of how I would apply the Singleton pattern for this class. Might not work / need some tweaking

Reply:
Is this an acceptable version @nortega-fh ?
`private static AppSettingsModel? _appSettings;

private AppSettingsInitializer()
{
    string appSettingsJson = File.ReadAllText(@"..\..\..\AppSettings\appsettings.json");
    _appSettings = JsonSerializer.Deserialize<AppSettingsModel>(appSettingsJson);
}

public static AppSettingsModel AppSettingsInstance()
{
    return _appSettings;
}`